app/controllers/certificates_controller.rb

```diff
@@ -1,6 +1,6 @@
 class CertificatesController < ApplicationController
-  before_action :set_certificate, except: [:index, :new]
-  before_action :set_volunteer, only: [:index, :new]
+  before_action :set_certificate, except: [:show, :index, :new, :create]
```

Comment:
Sorry, I accidentally deleted @manufaktor comment. 
The comment was:
> `:create` seems wrong here, since set_certificate tries to find a Certificate which isn't created yet

Reply:
So this could be written `only: [:edit, :update, :destroy]`, as well